Capstone project

RileyManda opened the pull request:
I have completed work on the Capstone Project according to the project [requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/capstone/business_requirements.md)

The following features have been implemented:

- The original design has been personalized with my own theme: Sapien Ai Creative Commons Global Summit 2023.
- The [design guidelines](https://www.behance.net/gallery/29845175/CC-Global-Summit-2015) :Colors,Typography,font face, size, and weight as well as the implementation of Lato font have been followed. 
- The pages have been developed to look almost identical to the original design with small adjustments to the text and image.
- Two pages have been built : The home page and The about page.
- Each of the pages has versions for 2 different screen sizes and has been built with a mobile-first approach and implemented desktop media queries.
- Links to the home page and about page have been included.
- The logo in the header links to the home page.

**Mobile menu**
- When the user clicks (or taps) the hamburger button on the header, the mobile menu appears over the page.
- The mobile menu has a close (X) button that closes the menu.
- The section "Featured speakers" has been created dynamically in JavaScript.


In addition, I have fulfilled the following requirements:

- Used semantic HTML tags.
- Used CSS selectors correctly.
- Used CSS box model.
- Use Flexbox to place elements in the page.
- Use images and backgrounds to enhance the look of a website.
- Is responsive  and utilizes media queries.
- Used GitHub Pages to deploy web pages.
- Used JavaScript to manipulate DOM elements.
- Used JavaScript events.
- Used objects to store and access data.

I have also ensured:
- There are [no linter errors](https://github.com/microverseinc/linters-config).
- Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
- Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- Follow the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

I have also included an attribution in the readme file for the images used in the past section.

I look forward to your feedback.

Thank you!
Best regards,

Shedrack-Sunday commented on index.html:
```diff
@@ -76,74 +88,10 @@ <h3>2023.9.12(THU) ~ 16(FRI)</h3>
 
         <!-- program section  START-->
         <section id="program-section">
-            <div class="program-container">
+            <div class="program-container" >
                 <h1>Main Program</h1>
                 <div class="section-indicator"></div>
```
- [ ] The `home` and `about` sections of your mobile view look clustered for the header section. 
- This can be fixed by placing appropriate styling to get the best possible margins and paddings to give a better user experience. Kindly see the comparisons with the original design below :

## Your work:
![image](https://github.com/RileyManda/conference-website/assets/50990925/1b8d1868-5616-4abb-9a32-92f55133b2cd)


## Design given on mobile:
![image](https://github.com/RileyManda/conference-website/assets/50990925/bfb1da6e-0f32-4aed-a986-b54df424241d)

Shedrack-Sunday commented on about.html:
```diff
@@ -129,23 +143,21 @@ <h1>Partner</h1>
+    <script type="module" src="./api/data.js"></script>
+    <script type="module" src="./api/about/logo-section.js"></script>
+    <script type="module" src="./api/about/summit.js"></script>
 
 </body>
 
```
- [ ] In the `footer` of the `About` section, kindly adjust the text to be aligned to the right just as the design is given, to have a close replica of what's given which perfects the experience of the end-user

## Your design:
![image](https://github.com/RileyManda/conference-website/assets/50990925/b871c9ce-a396-4644-ba7c-b168c971358c)

## Design given:
![image](https://github.com/RileyManda/conference-website/assets/50990925/227e5e08-f677-4313-a1dd-4b05053d693a)

Shedrack-Sunday commented on index.html:
```diff
@@ -179,19 +127,101 @@ <h3 class="speaker-card-title">SohYeong Noh</h3>
+    
+        </div>
+    </section>
+    <!-- partner section  START-->
+<section id="footer-section">
+    <div class="footer-container">
+        <div class="footer-column">
+            <img src="./assets/images/logo.svg" alt="">
+            <p>2023 Creative Commons Korea.<span>Some Rights Reserved.</span></p>
+        </div>
```
- [ ] The desktop version of the home has a white background. This design is missing in yours.
- Kindly add it up to have the exact replica of what's given 👍 

## Your design:
![image](https://github.com/RileyManda/conference-website/assets/50990925/40333788-a4da-4b22-ab0f-5a0d87e59b3b)

## Design given:
![image](https://github.com/RileyManda/conference-website/assets/50990925/44ee155f-a5e4-4722-a313-60e3e8b66e5a)

Shedrack-Sunday commented on index.html:
```diff
@@ -76,74 +88,10 @@ <h3>2023.9.12(THU) ~ 16(FRI)</h3>
         <!-- program section  START-->
         <section id="program-section">
-            <div class="program-container">
+            <div class="program-container" >
                 <h1>Main Program</h1>
```
- [ ] In the main program section of your homepage, you're missing th appropriate color, font and design given in general.
- Also, you didn't implement the hover effect on each of the cards.
- Kindly adjust this section to have a gray background with the right tet color and appropriate hover effect

## Your design:
![image](https://github.com/RileyManda/conference-website/assets/50990925/1f7613a1-17c8-4dcf-bd82-966c666d8e71)

## Design given:
![image](https://github.com/RileyManda/conference-website/assets/50990925/4c91e6eb-9bd7-4736-b0ab-26dfb576a7f0)

Shedrack-Sunday replied:
- [ ] Using heading tags (hx) instead of the multiple `h1` tag for headers is more semantically appropriate as it conveys the hierarchy and importance of the content, improving accessibility and SEO

christianbiring1 replied:
- [ ] Great work on implementing the `Main-program` section as required by the previous reviewer, However as you notice, you just give it a huge height compared to the original design, which should not suppose to be the case. So you may consider adjusting the size and make it look more similar to the original template:
- [ ] Also make sure we have the button `SEE THE WHOLE PROGRAM` as is the case on the original template.
### Your design :point_down: 
![image](https://github.com/RileyManda/conference-website/assets/97100091/989709fc-e52f-4e17-87f4-dbb2554f2712)

### Original design :point_down: 
![image](https://github.com/RileyManda/conference-website/assets/97100091/793cd8d7-5d55-462a-aeef-77a4adfde723)

christianbiring1 replied:
- [ ] Kindly make sure you use semantic tags for better accessibility by search Engines on your page as specified by the previous reviewer, `h1` is supposed to be unique on the whole page, other headings tags can be used to some other subtitles. you may use `h2...h6` here according to the structure of your page

Shedrack-Sunday commented on index.html:
```diff
@@ -8,28 +8,40 @@
     <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato">
     <link rel="stylesheet" href="style.css">
-    <title>Conference Website</title>
+    <title>SapienAi</title>
 </head>
 
 <body>
     <!-- navigation section| header -->
     <header class="toolbar">
+    <div class="social-navbar">
```
- [ ] You did well here 👏 . However, the experience of the user is the first thing that comes into mind when making the right design to yield success.
- Kindly adjust the logo to be in alignment with the header text, which has a very sync look to show details. 
- This will make this section look good and help you stick to the replica of the design which is the aim of the project 

## Your design:
![image](https://github.com/RileyManda/conference-website/assets/50990925/2a2fd9e4-11d4-43ba-a2ea-8ede7cb8a558)

## Design given:
![image](https://github.com/RileyManda/conference-website/assets/50990925/febae08c-65b5-4715-926f-ea8062debf1f)

Shedrack-Sunday commented on about.html:
```diff
@@ -77,35 +89,26 @@ <h3>Please contact us per Email for nay further questions about CC Global Summit
             <section id="creative-section">
                 <div class="creative-container">
                     <h1>Creative Commons
-                    Global Summit 2015 Logo</h1>
+                    Global Summit 2023 Logo</h1>
```
- [ ] The reason for having only one `h1` tag in a project is to maintain proper document structure and hierarchical order. The `h1` tag is the most important heading element in HTML and represents the main heading of a page or section. Having multiple `h1` tags can cause confusion in search engine optimization (SEO) and accessibility, as well as make it difficult for screen readers to understand the main content of the page. It is recommended to use only one h1 tag per page, and use other heading tags (h2, h3, etc.) to create a clear and structured document outline.

christianbiring1 replied:
- [ ] As requested in the previous review, Kinldy make sure you use the semantic tag in your markup, you may consider using any other heading element following the hierarchy as specified in the previous explanation. You may consider using any other `heading` element like `h2...h6` according to how your page is structured

Shedrack-Sunday commented on index.html:
```diff
@@ -179,19 +127,101 @@ <h3 class="speaker-card-title">SohYeong Noh</h3>
-                        MORE
-                        <i class="fas fa-chevron-down"></i>
-                    </a>
+                <div class="speaker-card more-speakers-lg">
+                    <img src="./assets/images/speaker_02.svg" alt="Lecture Icon">
+                    <div class="text-content">
+                        <h3 class="speaker-card-title">SohYeong Noh</h3>
+                        <p class="speaker-credential">Director of Art Centre Nabi and a board member of CC Korea</p>
+                        <div class="content-divider"></div>
+                        <p class="speaker-description">As the main venue for new media art production in Korea, Nabi
+                            promotes cross-disciplinary collaboration and
+                            understanding among science technology, humanities, and the arts.</p>
+                
+                
+                    </div>
+                
+                
                 </div>
 
+                <div class="speaker-card more-speakers-lg">
+                    <img src="./assets/images/speaker_02.svg" alt="Lecture Icon">
+                    <div class="text-content">
+                        <h3 class="speaker-card-title">SohYeong Noh</h3>
```
- [ ] Commented HTML code should be removed because it makes the codebase cluttered, increases file size, and may expose sensitive information or confuse other developers. Removing commented code ensures cleaner, optimized, and more secure web pages that are easier to maintain and troubleshoot.

Shedrack-Sunday reviewed (changes requested):
Hello @RileyManda         👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in the project,  However, there are a few you need to work on, to move to the next project. 💯

 
![](https://user-images.githubusercontent.com/80612925/157397398-00904e26-60ec-46cd-9510-240661c250ef.gif)

## Highlights   
- Good use of GitHub features ✔️    
- Very detailed PR ✔️ 

  
# Required Changes  🌵 

- [x] You made some impressive implementations which I must say is a job well done. However, there are a few features you missed or need to adjust to complement the brilliant work you've done so far 👏 


_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

RileyManda commented:
I have completed iterations on the Capstone Project as requested by the precious reviewer:

The following iterations have been made to replicate the design as requested by the reviewer:

- Aligned footer logo and text to match UX design footer logo and text margins
- Changed the background of the website to white to match the UX design
- Aligned header logo with header section content
- Implemented additional media queries for smaller screen sizes for responsiveness
- Removed additional h1 tags and used hx [h1,h2,to 3 incrementally]
- Performed a code cleanup of commented-out code
- Changed the background color and added some transparency to the Main program cards
- Added a hover effect to the Main program cards

In addition:

- I have updated the logo in desktop view to display a white logo in the back background.
- Designed the website according to the project [requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/capstone/business_requirements.md) to similar structure as the design with a custom theme.
- The original design has been personalized with my own theme: Sapien Ai Creative Commons Global Summit 2023.
- The [design guidelines](https://www.behance.net/gallery/29845175/CC-Global-Summit-2015) :Colors,Typography,font face, size, and weight as well as the implementation of Lato font have been followed.
- The pages have been developed to look almost identical to the original design with small adjustments to the text and image.
- Two pages have been built : The home page and The about page.
- Each of the pages has versions for 2 different screen sizes and has been built with a mobile-first approach and implemented desktop media queries.
- Links to the home page and about page have been included.
- The logo in the header links to the home page.

**Mobile menu**

- When the user clicks (or taps) the hamburger button on the header, the mobile menu appears over the page.
- The mobile menu has a close (X) button that closes the menu.
- The section "Featured speakers" has been created dynamically in JavaScript.

In addition, I have ensured:

- There are [no linter errors](https://github.com/microverseinc/linters-config).
- Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
- Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- Follow the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
- I have also included an attribution in the readme file for the images used in the past section.

I look forward to your feedback.

Thank you!
Best regards,

christianbiring1 commented on index.html:
```diff
@@ -152,46 +100,56 @@ <h3 class="card-title">CC Ignite</h3>
             <div class="speaker-container">
```
[OPTIONAL]
- [ ] You may consider adding some padding on the left and right of your speaker section to stick more to the design as you may notice on the desktop size your spacing is a bit small
### Original design :point_down: 
![image](https://github.com/RileyManda/conference-website/assets/97100091/744cda37-86a4-4eb9-bae4-4defb40821ac)
### Your design :point_down: 
![image](https://github.com/RileyManda/conference-website/assets/97100091/e8375f51-4e5a-41f3-acec-03c1d8110eee)

christianbiring1 reviewed (changes requested):
Hi @RileyManda , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
# STATUS: REQUEST CHANGES :recycle: 
![Uploading soclose.gif…]()


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

AbayomiOlaoye commented on about.html:
```diff
@@ -46,7 +58,7 @@
             <ul class="mobile-menu-list">
                 <li class="mobile-menu-list-items" onclick="closeView()"><a href="index.html">Home</a></li>
-                <li class="mobile-menu-list-items" onclick="closeView()"><a href="about.html">About</a></li>
+                <li class="mobile-menu-list-items active-item" onclick="closeView()"><a href="about.html">About</a></li>
```
- [ ] [OPTIONAL] Great efforts :muscle: but writing `JavaScript` in `HTML` files is actually not a good practice as professionals, and this is why we have the opportunity to break them up into different files and then link them to the HTML file using `script src=""`. You may consider implementing this event listener from your JS file :thinking:

AbayomiOlaoye commented on index.html:
```diff
@@ -76,122 +88,71 @@ <h3>2023.9.12(THU) ~ 16(FRI)</h3>
-                        </div>
-                    </div>
-
-                    <button class="program-btn">Join CC Global Summit</button>
-                </div>
-
+                <div class="program-content"></div>
+           
```
- [ ] Hi @RileyManda , like the previous reviewer mentioned, this main section is still taking up too much space, and after checking I discovered that it has a height of `100vh` which is really too much.
- See it here :point_down: 
![image](https://github.com/RileyManda/conference-website/assets/99715640/8df5ed3d-713c-4611-85b6-c102146672a0)

- Kindly remove or reduce the height to have a more user-friendly interface :pray:

AbayomiOlaoye commented on index.html:
```diff
@@ -38,14 +50,14 @@
             </div>
             <!-- Content of the mobile menu -->
             <ul class="mobile-menu-list">
-                <li class="mobile-menu-list-items" onclick="closeView()"><a href="index.html">Home</a></li>
+                <li class="mobile-menu-list-items active" onclick="closeView()"><a href="index.html">Home</a></li>
                 <li class="mobile-menu-list-items" onclick="closeView()"><a href="about.html">About</a></li>
             </ul>
```
- [ ] [OPTIONAL] The menu list renders (drops down) immediately screen is less than 768px without the user having to click on it. I think fixing this will improve your design, showing your true passion for accessibility :fire: 
- Just to mention, the nav--list is not complete on mobile views - just `home` and `about`, with very little focus in terms of font-size, positioning, and spacing. I might make more appeal if you fix all these 0 even though they are not compulsory.

AbayomiOlaoye reviewed (changes requested):
## STATUS: CHANGE REQUIRED ♻️ 

Hi @RileyManda :wave: :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![so_close](https://camo.githubusercontent.com/0c690ab997907a8c79afe7e695c891f6398cdbef93eaf8ce19f652ff321b54f8/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f744b797263304e546c4563475764546870342f67697068792e676966)


 ## Highlights
- Fairly nice design ✅ 
- Good use of semantic tags ✅ 
- Really nice concept (Ai) ✅ 

 ## Required Changes ♻️

- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
- You can also look into this, as it concerns accessibility and good design:
![image](https://github.com/RileyManda/conference-website/assets/99715640/e6a5f831-4c71-410b-b0ee-d9b38d73e37f)

  - It appears the font sizes for the heading, navigation, size of logo are too small to be visible to all potential users
  - Also, there is to much spacing before your main heading, and the line-heights might be more okay if increased 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

TuyishimireEric reviewed (approved):
# Status: APPROVED 🟢  
### Hi @RileyManda 👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
![done](https://media.giphy.com/media/l1uga8ObPBnIBoXUQ/giphy.gif)

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉

## Highlights
✔ Descriptive PR👍
✔ Made the suggested changes👍
✔ Professional README👍
✔ Awesome design👍
### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
